test/e2e/events/events.go

```diff
@@ -0,0 +1,200 @@
+package events
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"time"
+
+	ingressnodefwiov1alpha1 "github.com/openshift/ingress-node-firewall/api/v1alpha1"
+	testclient "github.com/openshift/ingress-node-firewall/test/e2e/client"
+	"github.com/openshift/ingress-node-firewall/test/e2e/daemonset"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type TestEvent struct {
+	InterfaceName      string
+	SourceAddress      string
+	DestinationAddress string
+	Action             ingressnodefwiov1alpha1.IngressNodeFirewallActionType
+	Protocol           ingressnodefwiov1alpha1.IngressNodeFirewallRuleProtocolType
+	// transport protocols only
+	DestinationPort string
+	// ICMP protocols
+	IcmpType int
+	IcmpCode int
+}
+
+func GetFromDaemonLogsOnNode(client *testclient.ClientSet, namespace, nodeName string, timeout time.Duration) ([]TestEvent, error) {
+	daemonsetPod, err := daemonset.GetDaemonSetOnNode(client, namespace, nodeName)
+	if err != nil {
+		return nil, err
+	}
+	request := client.CoreV1Interface.Pods(namespace).GetLogs(daemonsetPod.Name, &corev1.PodLogOptions{
+		Container: "events",
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	daemonsetLogsStream, err := request.Stream(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer daemonsetLogsStream.Close()
+
+	buffer := new(bytes.Buffer)
+	_, err = io.Copy(buffer, daemonsetLogsStream)
+	if err != nil {
+		return nil, err
+	}
+
+	return extractEventsFromString(buffer.String())
+}
+
+func extractEventsFromString(str string) ([]TestEvent, error) {
+	testEvents := make([]TestEvent, 0)
+	// Transport regular expression
+	// Match for each drop.
+	transportEventRE := "ruleId\\s([0-9]+)\\saction\\s(?P<action>\\w+).*if\\s(?P<inf>\\w+)\n.*\\ssrc\\saddr\\s(?P<srcaddr>[0-9.:a-z]+)\\sdst\\saddr\\s(?P<dstaddr>[0-9.:a-z]+)\n.*(?P<proto>tcp|udp|sctp)\\ssrcPort\\s\\d+\\sdstPort\\s(?P<dstport>\\d+)"
```

Comment:
lets at least just add a comment here and below for icmp case, in case the event strings changed in `pkg/ebpf/ingress_node_firewall_events.go` then this match expression need to be changed 
unfortuntely can't think of a good way to tie them up such that if the string in event changes the expression will get updated auto magically